vulture_os/toolkit/api_parser/messagetrace_o365/messagetrace_o365.py

```diff
@@ -0,0 +1,146 @@
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Vulture OS is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with Vulture OS.  If not, see http://www.gnu.org/licenses/.
+"""
+__author__ = "Robin Langlois"
+__credits__ = []
+__license__ = "GPLv3"
+__version__ = "4.0.0"
+__maintainer__ = "Vulture OS"
+__email__ = "contact@vultureproject.org"
+__doc__ = 'MessageTrace O365 API Parser'
+__parser__ = 'MESSAGETRACE O365'
+
+
+from datetime import timedelta
+import json
+import logging
+import requests
+
+from django.conf import settings
+from django.utils import timezone
+from toolkit.api_parser.api_parser import ApiParser
+import msal
+
+logging.config.dictConfig(settings.LOG_SETTINGS)
+logger = logging.getLogger('api_parser')
+
+
+class MessageTraceO365APIError(Exception):
+    pass
+
+
+class MessageTraceO365Parser(ApiParser):
+    def __init__(self, data):
+        super().__init__(data)
+
+        self.messagetrace_o365_client_id = data.get('messagetrace_o365_client_id')
+        self.messagetrace_o365_tenant_id = data.get('messagetrace_o365_tenant_id')
+        self.messagetrace_o365_client_secret = data.get('messagetrace_o365_client_secret')
+
+        self.grant_type = "client_credentials"
+
+        self.authority = f"https://login.microsoftonline.com/{self.messagetrace_o365_tenant_id}"
+        self.scope = ["https://outlook.office365.com/.default"]
+
+        self.url = "https://reports.office365.com/ecp/reportingwebservice/reporting.svc/MessageTrace"
+
+    def __connect(self):
+        logger.info(f"[{__parser__}]:connect: Requesting a new token", extra={'frontend': str(self.frontend)})
+        app = msal.ConfidentialClientApplication(
+            self.messagetrace_o365_client_id, authority=self.authority, client_credential=self.messagetrace_o365_client_secret
+        )
+        token_result = app.acquire_token_for_client(scopes=self.scope)
+        if "error" in token_result:
+            raise MessageTraceO365APIError("Could not retrieve token : " + token_result["error_description"])
+        self.access_token = token_result.get("access_token")
+
+        self.headers = {
+            "Authorization": f"Bearer {self.access_token}",
+            "Accept": "application/json"
+        }
+
+    def _get_logs(self, since, to, limit=None):
+        self.__connect()
+        filter_str = f"StartDate eq datetime'{since}' and EndDate eq datetime'{to}'"
+
+        # without limit, max 2000 results
+        params = {
+            "$filter": filter_str,
+            "$top": limit
+        }
+
+        response = requests.get(self.url,
```

Comment:
Add an `info` logger with request details (params) for debugging (but be warry to omit confidential info, such as access tokens)